store/store.go

```diff
@@ -467,6 +460,20 @@ func (s *Store[H]) flush(ctx context.Context, headers ...H) error {
 	return batch.Commit(ctx)
 }
 
+func (s *Store[H]) loadHeadKey(ctx context.Context) error {
+	h, err := s.readHead(ctx)
+	if err != nil {
+		return err
+	}
+
+	newHeight := s.advanceContiguousHead(ctx, h.Height())
+	if newHeight >= h.Height() {
+		s.contiguousHead.Store(&h)
```

Comment:
wouldn't this already happen in `updateContiguousHead`?

Reply:
It might not happen if we will not find a higher head due to `if currHeight > prevHeight {`.

That's why we do here `>=` and  update.

Reply:
Why can't this check be de-duplicated and left only to `advanceContiguousHead` where it can check for `>=` on #L541 ?

Reply:
Well, can be. The only things is that we will do a bit more often `headKey` updates. I don't see a problem with that, done.

Reply:
Ah, no. We need a real header (`newHead`) to do `updateContiguousHead`. Adding `>=` will complicate code for not a little reason. Reverted.

Reply:
Can be resolved?

Reply:
I still do not understand why we are doing this check, in different formats, in 2 different places. It's complicated.

Reply:
After today's changes this can be simplified even more. Done.